mitosheet/mitosheet/mito_widget.py

```diff
@@ -142,11 +143,18 @@ def handle_update_event(self, event: Dict[str, Any]) -> None:
         - Name of an existing analysis
         """
 
-        self.steps_manager.handle_update_event(event)
-
-        # Update all state variables
-        self.update_shared_state_variables()
+        try:
+            self.steps_manager.handle_update_event(event)
 
+            # Update all state variables
+            self.update_shared_state_variables()
+        except:
+            # We handle the case of replaying the analysis specially, because we don't
```

Comment:
Update this comment to explain, and that we include the updating of the shared state variables!